API/Services/MetadataService.cs

```diff
@@ -10,7 +10,7 @@ namespace Backend.API.Services
 {
     public interface IMetadataService
     {
-        public Task<StoredMetadata> GetMetadata(string name);
+        public Task<StoredMetadata> GetMetadata(string datasourceId);
```

Comment:
Can we make the `GetMetadata` parameter be "properly" typed by having the type be `DatasourceId`?